tests/unit/test_group_project.py

```diff
@@ -230,6 +235,22 @@ def test_peer_grade_questions(self, expected_result, *stage_questions):
 
             get_stages.assert_called_with(PeerReviewStage.CATEGORY)
 
+    @ddt.data(
+        (None, []),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', '2016-02-29 04:30:48', '2016-10-29 04:30:48']),
```

Comment:
@ziafazal I'm being a little bit pedantic here, but test cases now can be satisfied by another implementation: always take open_date of 3rd (or last) stage. So, it would be great to cover the following scenarios:

1. Some stages does not have open date (i.e. open_date = None)
2. All stages does not have open date
3. Just normal list with max date in some other place (first, in the middle)
4. Edge cases: two equal dates, two equal + third larger, two equal + third lower, minor differences (i.e. seconds)